__tests__/app.test.js

```diff
@@ -131,4 +131,42 @@ describe('api/articles', () => {
     .get('/api/notarticles')
     .expect(404)
   })
+})
+
+describe('api/articles/:article_id/comments', () => {
+  test('GET 200: returns an array of comments by article ID', () => {
+    return request(app)
+    .get(`/api/articles/9/comments`)
+    .expect(200).then(({body})=> {
+      const commentArray = body.comments
+
+      expect(Array.isArray(commentArray)).toBe(true) 
+
+      commentArray.forEach(comment => {
+        expect(comment).toHaveProperty('comment_id')
+        expect(comment).toHaveProperty('article_id')
+        expect(comment).toHaveProperty('body')
+        expect(comment).toHaveProperty('created_at')
+        expect(comment).toHaveProperty('author')
+        expect(comment).toHaveProperty('votes')
+      })
```

Comment:
When we do assertions using loops we run the risk of an empty array creating a false positive in the tests (no items in the arr means the loop doesn't run which means the assertions inside the loop don't fail). So alongside a loop we always want to assert that the array actually contains items or the right num of items